gpflow/base.py

```diff
@@ -59,17 +60,35 @@ def __init__(self,
             self._unconstrained = tf.Variable(value, dtype=dtype, name=name, trainable=trainable)
 
         self.prior = prior
+        self.prior_on_constrained = prior_on_constrained
         self._transform = transform
 
-    def log_prior(self):
-        x = self.read_value()
-        y = self._unconstrained
+    def log_prior(self, evaluate_on_constrained: bool = True):
```

Comment:
This is the line that changes the behaviour of GPflow, correct? I.e. "backwards-compatible" would be `prior_on_constrained=True` but `evaluate_on_constrained=False`. If we want to change this, now (transition to gpflow-2.0) is probably the right time, just want to make sure we advertise this sufficiently loudly.

Reply:
Yes exactly, if we set this to False then we recover the previous behaviour